### DISCUSSION TITLE
Montage and annotate - annotation is over all montage images

### DISCUSSION THREAD

**Discussion by tomtomhoward** [ImageMagick/ImageMagick#7072] [Help] [answered]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

macOS via Homebrew

### Description

Hi,
I'm trying to join three images together, and write something over the last image, or even the space to the right of the last image in the padding.
When I use -annotate or -draw in a montage, it draws the text over all 3 images, so it appears in every frame of the montage.

My current command is....

montage 1.jpg 2.jpg 3.jpg -geometry 600x600^+100+100 -tile x1 -fill black -pointsize 200 -gravity east  -draw "text 0,0 'hello world'" montage.jpg

Do I need to do something different eg montage it, then add the -draw as second process, or draw the text over the last frame first then montage? Can it be done in one command?

**Comment by snibgo** [comment, marked as answer]
You could write the output of the `magick montage` command to `miff:-` for miff format to stdout, and pipe that to another `magick` that does whatever you want.
```
magick montage 1.jpg 2.jpg 3.jpg -geometry 600x600^+100+100 -tile x1 miff:- | magick - -fill black -pointsize 200 -gravity east -draw "text 0,0 'hello world'" montage.jpg
```

**Comment by tomtomhoward** [comment]
Cheers that works fine and keeps the file handling tidy - I assume then there isn't a way to run it in one?
I might actually do that the other way round and write the text over the last image and then pipe the miff into the montage - is the single hypen the placeholder you could use and put later in the argument?

What I'm doing is combining two photos and an event logo on the end - to automatically generate photo strips in a Photo Booth from multiple photos - and I want to write the datestamp under the logo discreetly somehow so I can tie it back to the file on the computer later. (ie if someone wants a 2nd print later on)

Later edit: 
Thanks so I have turned that round;

`magick 3.jpg -fill black -pointsize 200 -gravity east -draw "text 0,0 'hello world'" miff:- | magick montage 1.jpg 2.jpg - -geometry 600x600^+100+100 -tile x1 montage.jpg`

And that works great for placing on the last time - although I couldn't place into the padding it makes positioning easier

**Comment by snibgo** [comment]
`magick montage` doesn't have the full flexibility of `magick`, but we can do this:
```
magick montage 1.jpg 2.jpg ( 3.jpg -fill black -pointsize 200 -gravity east -draw "text 0,0 'hello world'" ) out.png
```
